lib/backend/syncersv1/updateprocessors/selectors.go

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2019 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package updateprocessors
+
+import (
+	"fmt"
+
+	"github.com/projectcalico/libcalico-go/lib/selector/parser"
+	log "github.com/sirupsen/logrus"
+)
+
+// ParseSelectorAttachPrefix takes a v3 selector and returns the appropriate v1 representation
+// by prefixing the keys with the given prefix.
+// If prefix is `pcns.` then the selector changes from `k == 'v'` to `pcns.k == 'v'`.
+func ParseSelectorAttachPrefix(s, prefix string) string {
```

Comment:
I think this and the function below should remain private - I don't see anywhere they're used outside of this package?

Reply:
Fixed